src/NewsService.Mappers/Models/NewsInfoMapper.cs

```diff
@@ -1,106 +1,41 @@
-        Response<IOperationResult<IGetUserDataResponse>> response = _userRequestClient.GetResponse<IOperationResult<IGetUserDataResponse>>(request).Result;
-
-        if (!response.Message.IsSuccess)
-        {
-          _logger.LogWarning($"Can't found user. Reason: '{string.Join(',', response.Message.Errors)}'");
-        }
-        else
-        {
-          fullName = $"{response.Message.Body.FirstName} {response.Message.Body.LastName} {response.Message.Body.MiddleName}".Trim();
-        }
-      }
-      catch (Exception exception)
-      {
-        _logger.LogError(exception, "Exception on get user data request.");
-      }
-
-      return fullName;
-    }
-
-    private string GetDepartmentName(Guid departmentId)
-    {
-      string name = null;
-
-      try
-      {
-        object request = IGetDepartmentRequest.CreateObj(null, departmentId);
-        Response<IOperationResult<IGetDepartmentResponse>> response = _departmentRequestClient.GetResponse<IOperationResult<IGetDepartmentResponse>>(request).Result;
-
-        if (!response.Message.IsSuccess)
-        {
-          _logger.LogWarning($"Can't found department name. Reason: '{string.Join(',', response.Message.Errors)}'");
-        }
-        else
-        {
-          name = response.Message.Body.Name;
-        }
-      }
-      catch (Exception exception)
-      {
-        _logger.LogError(exception, "Exception on get department data request.");
-      }
-
-      return name;
-    }
+    private readonly IUserInfoMapper _userInfoMapper;
+    private readonly IDepartmentInfoMapper _departmentInfoMapper;
 
     public NewsInfoMapper(
-      IRequestClient<IGetUserDataRequest> userClient,
-      IRequestClient<IGetDepartmentRequest> departmentClient,
-      ILogger<NewsInfoMapper> logger)
+      IUserInfoMapper userInfoMapper,
+      IDepartmentInfoMapper departmentInfoMapper)
     {
-      _userRequestClient = userClient;
-      _departmentRequestClient = departmentClient;
-      _logger = logger;
+      _userInfoMapper = userInfoMapper;
+      _departmentInfoMapper = departmentInfoMapper;
     }
 
-    public NewsInfo Map(DbNews dbNews)
+    public NewsInfo Map(DbNews dbNews, List<DepartmentData> departments, List<UserData> authors, List<ImageData> avatarImage)
     {
       if (dbNews == null)
       {
         return null;
       }
 
-      Department department = null;
-      if (dbNews.DepartmentId != null && dbNews.DepartmentId != Guid.Empty)
-      {
-        department = new Department { Id = (Guid)dbNews.DepartmentId, Name = GetDepartmentName((Guid)dbNews.DepartmentId) };
-      }
-
       return new NewsInfo
       {
         Id = dbNews.Id,
         Preview = dbNews.Preview,
         Subject = dbNews.Subject,
-        Author = new User { Id = dbNews.AuthorId, FullName = GetUserFullName(dbNews.AuthorId) },
-        Department = department,
+        Departments = departments?.Select(_departmentInfoMapper.Map).ToList(),
```

Comment:
better to conver department info before this map, because news can have same departments, so for each news you will be convert one DepartmentData to Info instead one time convert all of them

Reply:
done